Python/PRP/ConditionalObject/pyConditionalObject.cpp

```diff
@@ -0,0 +1,153 @@
+/* This file is part of HSPlasma.
+ *
+ * HSPlasma is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * HSPlasma is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with HSPlasma.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <PyPlasma.h>
+#include <PRP/ConditionalObject/plConditionalObject.h>
+#include "pyConditionalObject.h"
+#include "PRP/KeyedObject/pyKeyedObject.h"
+#include "PRP/pyCreatable.h"
+
+static inline plConditionalObject* IConvertCond(pyConditionalObject* self) {
+    return plConditionalObject::Convert(IConvert((pyCreatable*)self));
+}
+
+extern "C" {
+
+static PyObject* pyConditionalObject_new(PyTypeObject* type, PyObject* args, PyObject* kwds) {
+    PyErr_SetString(PyExc_RuntimeError, "plConditionalObject is abstract");
+    return NULL;
+}
+
+static PyObject* pyConditionalObject_getSatisfied(pyConditionalObject* self, void*) {
+    return PyBool_FromLong(IConvertCond(self)->getSatisfied() ? 1 : 0);
```

Comment:
With VC++, if you try to access fThis by casting self to anything but its final type, there is an access violation. IIRC Branan came up with this solution many moons ago. It's all over PyHSPlasma, I just chose to shorten things a bit here because the long version is just silly.